geopandas/_vectorized.py

```diff
@@ -67,9 +71,15 @@ def _pygeos_to_shapely(geom):
     if geom is None:
         return None
 
+    if compat.USE_SHAPELY_20:
+        return geom
+
     if compat.PYGEOS_SHAPELY_COMPAT:
-        geom = shapely.geos.lgeos.GEOSGeom_clone(geom._ptr)
-        return shapely.geometry.base.geom_factory(geom)
+        # we can only use this compatible fast path for shapely < 2, because
+        # shapely 2+ doesn't expose clone
+        if not compat.SHAPELY_GE_20:
```

Comment:
Should `PYGEOS_SHAPELY_COMPAT` also be set to `False` where pygeos and shapely >= 2.0 are also present?  In which case, we don't need this extra conditional, but good to retain the comment somewhere.

Reply:
We can still convert pygeos<->shapely geometries using the faster path when the GEOS versions are compatible, also if Shapely 2.0 is installed?

Reply:
> But I thought (per the comment in code) that the faster path wasn't available due to `clone' not being available - are you suggesting adding that to shapely 2.0?

Ah, yes, that's correct. But we also use this variable for the shapely->pygeos conversion, and that direction can still make use of the fast-path.